internal/common/common.go

```diff
@@ -150,8 +151,25 @@ func GetNetworkCidrAttr(obj interface{}, fieldName string) []*string {
 	}
 
 	funk.ForEach(field, func(elem interface{}) {
-		addresses = append(addresses, swag.String(string(funk.Get(elem, "Cidr").(models.Subnet))))
+		address := funk.Get(elem, "Cidr")
+		if address != nil {
+			addresses = append(addresses, swag.String(string(address.(models.Subnet))))
+		}
 	})
 
 	return addresses
 }
+
+// IsNetworksSliceNonEmpty checks whether the provided slice of networks, i.e. objects containing
+// CIDR is non-empty. The slice is assumed to be non-empty if at least one of its elements
+// contains a non-empty CIDR.
+func IsNetworksSliceNonEmpty(arg interface{}) bool {
```

Comment:
To some extent, it seems like this is something that we should validate at the source or have specific types/functions to validate `MachineNetworks` and `ClusterNetworks`.

One option could be to have an interface with a `HasCIDR` / `IsEmpty` function, which both `MachineNetworks` and `ClusterDeployment` implement. This should provide the same results in a hopefully more maintainable/easier-to-follow way. 

Not opposed to the solution in this PR, just throwing another idea out there :)